geopandas/plotting.py

```diff
@@ -36,12 +41,15 @@ def _flatten_multi_geoms(geoms, colors):
         else:
             components.append(geom)
             component_colors.append(color)
+
+    if was_none:
+        component_colors = None
```

Comment:
yes, matplotlib would raise on such data anyway, so then I can probably use that check